test/bin/mdx-pp/misc/compile/dune

```diff
@@ -0,0 +1,10 @@
+(executable
```

Comment:
Why an extra `misc/` level ? `compile` would fit near `expect` in the parent directory, I think.

Reply:
For the same reasons I mentioned above, I think it's better if all test case folders have the same "depth" in the folder structure, that way you don't have to try and interpret the names of the folders in `test/bin/mdx-pp/` to figure out which one are used to group similar test cases and which ones are individual test cases.

Reply:
I see, makes sense.